src/Search/SearchResults.php

```diff
@@ -164,13 +164,13 @@ public function __construct(PagerfantaInterface $pagerFantaResults, SearchOption
      */
     private function processResults(): void
     {
-        $this->currentPage = $this->searchOptions->getCurrentPage();
-        $this->pagerFantaResults->setCurrentPage($this->currentPage);
         $this->pagerFantaResults->setMaxPerPage($this->searchOptions->getMaxPerPage());
 
         $this->numberOfResults = $this->pagerFantaResults->getNbResults();
         $this->numberOfPages = $this->pagerFantaResults->getNbPages();
         $this->resultsPerPage = $this->pagerFantaResults->getMaxPerPage();
+        $this->currentPage = ($this->searchOptions->getCurrentPage() <= $this->numberOfPages) ? $this->searchOptions->getCurrentPage() : 1;
+        $this->pagerFantaResults->setCurrentPage($this->currentPage);
```

Comment:
You should still set current page to 1

Reply:
I am setting $this->currentPage = 1 right here. What do you mean?

Reply:
most have been looking at old code, NM